### ISSUE TITLE
False positive for AL05 when using a wildcard select passed to a function

### ISSUE THREAD

**Issue by reteps** [sqlfluff/sqlfluff#7235]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Similar to #7234, `aliasing.unused` doesn't consider wildcards being passed to functions.

```
== [/Users/peter/work/PrairieLearn/apps/prairielearn/src/lib/question-render.sql] FAIL
L: 210 | P:  33 | AL05 | Alias 'lgj' is never used in SELECT statement.
                       | [aliasing.unused]
```

### Expected Behaviour

A used table alias does not flag `AL05`.

### Observed Behaviour

A used table alias does flag `AL05`.

### How to reproduce

This query fails lint, and should not flag, even though the alias is used:

```sql
SELECT
  to_jsonb(mot.*)
FROM
  my_table
  JOIN my_other_table AS mot ON (TRUE);
```

This query also fails lint:

```sql
SELECT
  to_jsonb(mot)
FROM
  my_table
  JOIN my_other_table AS mot ON (TRUE);
```

This query works:

```sql
SELECT
  to_jsonb(mot.column)
FROM
  my_table
  JOIN my_other_table AS mot ON (TRUE);
```

And so does this one:

```sql
SELECT
  mot.*
FROM
  my_table
  JOIN my_other_table AS mot ON (TRUE);
```

### Dialect

Postgres

### Version

sqlfluff, version 3.5.0, Python 3.13.5

### Configuration

```
[sqlfluff]
templater = placeholder
dialect = postgres
large_file_skip_byte_limit = 0
sql_file_exts = .sql
exclude_rules = aliasing.column,
    aliasing.expression,
    aliasing.table,
    aliasing.unique.table,
    # We are fine with using SELECT * in queries.
    ambiguous.column_count,
    ambiguous.join,
    # TODO: Consider enabling this.
    ambiguous.order_by_limit,
    capitalisation.functions,
    capitalisation.identifiers,
    capitalisation.types,
    convention.casting_style,
    layout.cte_newline,
    layout.functions,
    layout.indent,
    layout.long_lines,
    layout.select_targets,
    layout.spacing,
    layout.start_of_file,
    references.consistent,
    references.from,
    references.keywords,
    references.qualification,
    structure.column_order,
    structure.distinct,
    # While explicit, this provides clarity.
    structure.else_null,
    structure.join_condition_order,
    structure.using

[sqlfluff:templater:placeholder]
param_style = dollar

# $foo is replaced by foo in the templater (no quotes), so we'll set them to something reasonable,
# like the name of the parameter in quotes, so that it can parse correctly.
start = 'start'
end = 'end'
offset = 'offset'
embedding = 'embedding'
enabled_authn_provider_ids = []
limit = 1000
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by jonatanschroeder** [comment]
I don't think this is fully completed. `TO_JSONB(lgj.*)` is now correctly tagged as a use, but `TO_JSONB(lgj)` still fails lint, even though they are equivalent.
